libraries/core/src/test/java/com/paritytrading/philadelphia/DoublePipeChannel.java

```diff
@@ -0,0 +1,66 @@
+package com.paritytrading.philadelphia;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ByteChannel;
+import java.nio.channels.GatheringByteChannel;
+import java.nio.channels.Pipe;
+import java.nio.channels.ScatteringByteChannel;
+
+// testing utility - in  prod code this should be selectable, and error handling more robust
+public class DoublePipeChannel implements ByteChannel, ScatteringByteChannel, GatheringByteChannel {
+    private final Pipe.SinkChannel sink;
+    private final Pipe.SourceChannel source;
+
+    public DoublePipeChannel(Pipe.SinkChannel sink, Pipe.SourceChannel source) {
+        this.sink = sink;
+        this.source = source;
+    }
+
+    void configureBlocking(boolean block) throws IOException {
+        sink.configureBlocking(block);
+        source.configureBlocking(block);
+    }
```

Comment:
As this class is only used in non-blocking mode, perhaps the constructor could directly set both channels into non-blocking mode? Then this method wouldn't be needed at all.